File: HumanitarianAssistance.Entities/Models/Marketing/PolicyModel.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace HumanitarianAssistance.ViewModels.Models.Marketing
{
    public class PolicyModel
    {
        public long PolicyId { get; set; }
        public string PolicyName { get; set; }
        public string PolicyCode { get; set; }
        public long? LanguageId { get; set; }
        public string LanguageName { get; set; }
        public long? ProducerId { get; set; }
        public string ProducerName { get; set; }
        public long? MediumId { get; set; }
        public string MediumName { get; set; }
        public long? MediaCategoryId { get; set; }
        public string MediaCategoryName { get; set; }
        public string CreatedById { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? CreatedDate { get; set; }
        public bool? IsDeleted { get; set; }
        public string ModifiedBy { get; set; }
        public string ModifiedById { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public string Description { get; set; }
    }

    public class PolicyFilterModel
    {
        public bool Medium { get; set; }
        public string Value { get; set; }
        public bool PolicyId { get; set; }
        public bool PolicyName { get; set; }
    }

    public class PolicyPaginationModel
    {
        public int pageIndex { get; set; }
        public int pageSize { get; set; }
    }
}

File: HumanitarianAssistance.WebApi/NewUI/src/app/dashboard/marketing/broadcast-policy/model/policy-model.ts
export interface PolicyModel {
  PolicyId?: number;
  PolicyName?: string;
  PolicyCode?: string;
  LanguageId?: number;
  LanguageName?: string;
  ProducerId?: number;
  ProducerName?: string;
  MediumId?: number;
  MediumName?: string;
  MediaCategoryId?: number;
  MediaCategoryName?: string;
  CreatedById?: string;
  CreatedBy?: string;
  CreatedDate?: any;
  IsDeleted?: boolean;
  ModifiedBy?: string;
  ModifiedById?: string;
  ModifiedDate?: any;
  Description?: string;
}



export interface FilterPolicyModel {
  Value?: string;
  Medium?: boolean;
  PolicyId?: boolean;
  PolicyName?: boolean;
}

export interface PolicyPaginationModel {
  pageIndex?: number;
  pageSize?: number;
}

export interface StartEndTimeModel {
  startTime?: any;
  endTime?: any;
}

export interface PolicyTimeModel {
  PolicyId?: number;
  RepeatDays?: any;
  StartTime?: string;
  EndTime?: string;
  Id?: number;
}

export interface PolicyDayModel {
  PolicyId?: number;
  Id?: number;
  Monday?: boolean;
  Tuesday?: boolean;
  Wednesday?: boolean;
  Thursday?: boolean;
  Friday?: boolean;
  Saturday?: boolean;
  Sunday?: boolean;
}

export interface PolicyOrderScheduleModel {
  PolicyId?: number;
  Id?: number;
  startDate?: Date;
  endDate?: Date;
}
